### ISSUE TITLE
Backgrounding app during FxA sign in drops user back to Welcome screen

### ISSUE THREAD

**Issue by devinreams** [mozilla-lockbox/lockbox-android#309]
Follow up to #240 

Backgrounding the app when in process of signing in (in the FxA web view) always "forgets" and takes me back to "Welcome" screen, not the in-progress webview

This is challenging and frustrating if the user has to switch apps or go find a password and come back to a blank "Welcome" state instead of their in-progress web form.

**Comment by sashei** [comment]
I haven't been able to replicate this issue _exactly_; however, I did find that when returning to the FxA login screen from the background (particularly with "Do not keep Activities" enabled) it will reload the entire webview, forcing you to enter your email again and having the same effect as what @devinreams describes above. I'm curious to hear what others are experiencing now that #260 is fixed @nickbrandt @changecourse @linuxwolf @eliserichards .

**Comment by linuxwolf** [comment]
@sashei I experienced what you describe; returning to foreground takes me straight to the WebView (yay) but I have to start over (boo).

**Comment by sashei** [comment]
OK, I'm assuming that the core issue was addressed with #260, but will repurpose this issue to reflect our findings about restarting the login flow on backgrounding with "Do not keep Activities" enabled.

**Comment by rfk** [comment]
Is this somethat that might need to be fixed on the FxA side of things, e.g. by remembering that you were partway through an OAuth flow and trying to restart from where you left off rather than starting it over from scratch?  (I've no idea how complex that might be, but it's worth asking).

**Comment by sashei** [comment]
Hm, I think it would be hard to differentiate on the FxA side between a few cases in which the user might start the flow and then back out -- some possibilities include:

- hitting the X or back button to return to the Welcome screen => my expectation would be that the FxA flow starts over
- backgrounding & foregrounding the app from the FxA screen => FxA resumes where you left off, even if the Activity got cleaned up in the background
- hard quitting from the the FxA screen and re-opening => the app will cold start on the Welcome screen in this scenario, my expectation is that the FxA flow will start over

@nickbrandt @changecourse @sandysage do y'all feel I've enumerated the behavior as you would expect it in the three scenarios above?

**Comment by sandysage** [comment]
@sashei That is correct. Thank you for accurately capturing the expectations. I'm pretty sure this is the behavior we have in iOS too.

**Comment by devinreams** [comment]
Downgrading due to the fact we encountered this since we were not keeping activity state.
